test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,349 @@
+    auto channelOperationResult = ChannelRequestHelper(channelOperation);
+
+    return channelOperationResult == S_OK;
+}
+
+bool MultipleChannelClose()
+{
+    auto channelOperation = PushNotificationManager::CreateChannelAsync(remoteId1);
+    if (channelOperation.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        channelOperation.Cancel();
+        return false; // timed out or failed
+    }
+
+    auto result = channelOperation.GetResults();
+    auto status = result.Status();
+    if (status != PushNotificationChannelStatus::CompletedSuccess)
+    {
+        return false; // did not produce a channel
+    }
+
+    result.Channel().Close();
+    try
+    {
+        result.Channel().Close();
+    }
+    catch (...)
+    {
+        auto channelRequestException = hresult_error(to_hresult());
+        return channelRequestException.code() == WPN_E_CHANNEL_CLOSED;
+    }
+    return false;
+}
+
+bool MultipleChannelRequestUsingSameRemoteId()
+{
+
+    auto channelOperation1 = PushNotificationManager::CreateChannelAsync(remoteId1);
+    auto channelOperation2 = PushNotificationManager::CreateChannelAsync(remoteId1);
+    auto channelOperationResult2 = ChannelRequestHelper(channelOperation2);
+    auto channelOperationResult1 = ChannelRequestHelper(channelOperation1);
+
+    return channelOperationResult2 == S_OK;
+}
+
+bool MultipleChannelRequestUsingMultipleRemoteId()
+{
+    auto channelOperation1 = PushNotificationManager::CreateChannelAsync(remoteId1);
+    auto channelOperation2 = PushNotificationManager::CreateChannelAsync(remoteId2);
+    auto channelOperationResult2 = ChannelRequestHelper(channelOperation2);
+    auto channelOperationResult1 = ChannelRequestHelper(channelOperation1);
+
+    return channelOperationResult2 == S_OK;
+}
+
+bool ActivatorTest()
+{
+    PushNotificationManager::UnregisterActivator(appToken, PushNotificationRegistrationOption::PushTrigger | PushNotificationRegistrationOption::ComActivator);
+    try
+    {
+        PushNotificationActivationInfo info(
+            PushNotificationRegistrationOption::PushTrigger | PushNotificationRegistrationOption::ComActivator,
+            c_fakeComServerId);
+
+        fakeToken = PushNotificationManager::RegisterActivator(info);
+        if (!fakeToken.TaskRegistration())
+        {
+            return false;
+        }
+
+        PushNotificationManager::UnregisterActivator(fakeToken, PushNotificationRegistrationOption::PushTrigger | PushNotificationRegistrationOption::ComActivator);
+    }
+    catch (...)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool RegisterActivatorNullDetails()
+{
+    winrt::hresult hr = S_OK;
+    try
+    {
+        PushNotificationManager::RegisterActivator(nullptr);
+    }
+    catch (...)
+    {
+        auto registerActivatorException = hresult_error(to_hresult());
+        hr = registerActivatorException.code();
+    }
+    return hr == E_INVALIDARG;
+}
+
+bool RegisterActivatorNullClsid()
```

Comment:
The tests which intentionally do invalid things need a comment to say "Verify that X is not allowed", so people won't copy/pasta this code thinking that it's supposed to work.

```cpp
// Verify that the null CLSID is rejected.
```

and then later

```cpp
    // This call is invalid and should fail.
    PushNotificationManager::RegisterActivator(info);
```